### DISCUSSION TITLE
[Proposal] read-only/immutable local declarations

### DISCUSSION THREAD

**Discussion by thomhurst** [dotnet/csharplang#8259] [Language Ideas] [answered]
Many times, we declare a variable so we can re-use the result, but we don't intend to ever change or overwrite that result. Sometimes we may accidentally do something to change it which can result in bugs.

If we don't intend to change the value, it makes sense to be able to declare it as immutable at a local level. This would then give us compiler errors if we try to reassign it.

This is common in other languages too.

Kotlin > Val
JavaScript > const 
Java > final
Etc.

**Comment by TahirAhmadov** [comment, marked as answer]
https://github.com/dotnet/csharplang/issues/188
